tokio/src/sync/mpsc/bounded.rs

```diff
@@ -152,6 +165,27 @@ impl<T> Receiver<T> {
 
     /// Blocking receive to call outside of asynchronous contexts.
     ///
+    /// This method returns `None` if the channel has been closed and there are
+    /// no remaining messages in the channel's buffer. This indicates that no
+    /// further values can ever be received from this `Receiver`. The channel is
+    /// closed when all senders have been dropped, or when [`close`] is called.
+    ///
+    /// If there are no messages in the channel's buffer, but the channel has
+    /// not yet been closed, this method will block until a message is sent or
+    /// the channel is closed.
+    ///
+    /// This method is intended for use cases where you are sending from
+    /// asynchronous code to synchronous code, and will work even if the sender
+    /// is not using [`blocking_send`] to send the message.
```

Comment:
This particular note is what inspired this PR, since whether the channels works when one end is sync and the other is async is a surprisingly common question.